crates/pixi_glob/src/glob_set.rs

```diff
@@ -0,0 +1,164 @@
+use std::{
+    io,
+    path::{Path, PathBuf},
+};
+
+use itertools::{Either, Itertools};
+use thiserror::Error;
+use wax::Glob;
+
+pub(crate) struct GlobSet<'t> {
+    /// The globs to include in the filter.
+    pub include: Vec<Glob<'t>>,
+    /// The globs to exclude from the filter.
+    pub exclude: Vec<Glob<'t>>,
+}
+
+#[derive(Error, Debug)]
+pub enum GlobSetError {
+    #[error("failed to access {}", .0.display())]
+    Io(PathBuf, #[source] io::Error),
+
+    #[error(transparent)]
+    DirWalk(#[from] io::Error),
+
+    #[error("failed to read metadata for {0}")]
+    Metadata(PathBuf, #[source] wax::WalkError),
+
+    #[error(transparent)]
+    Build(#[from] wax::BuildError),
+}
+
+pub(crate) struct MatchedFile {
+    /// Path to the matched file
+    pub matched_path: PathBuf,
+    /// Metadata of the matched file
+    pub metadata: std::fs::Metadata,
```

Comment:
Im not entirely sure but I think accessing the metadata is another filesystem operation (separate from opening the file itself) which especially on Windows might incur a non-significant overhead. Given that we don't need it for computing sha hashes that might be wasteful.